asyncstdlib/heapq.py

```diff
@@ -145,7 +145,9 @@ async def merge(
     is yielded before ``b``. Use ``reverse=True`` for descending sort order.
     The ``iterables`` must be pre-sorted in the same order.
     """
-    a_key = awaitify(key) if key is not None else None
+    a_key: Callable[[Any], Awaitable[Any]] | None = (
```

Comment:
For consistency with other annotations, please use `Optional[…]` instead of `… | None` for now.
```suggestion
    a_key: Optional[Callable[[Any], Awaitable[Any]]] = (
```